src/main/java/org/humancellatlas/ingest/biomaterial/web/BiomaterialController.java

```diff
@@ -77,4 +94,93 @@
         PersistentEntityResource resource = assembler.toFullResource(updatedBiomaterial);
         return ResponseEntity.accepted().body(resource);
     }
+
+    @RequestMapping(path = "/biomaterials/{id}/inputToProcesses", method = {PUT, POST}, consumes = {TEXT_URI_LIST_VALUE})
+    HttpEntity<?> overrideLinkBiomaterialAsInputToProcessesDefaultEndpoint(@PathVariable("id") Biomaterial biomaterial,
+                                                                           @RequestBody Resources<Object> incoming,
+                                                                           HttpMethod requestMethod,
+                                                                           PersistentEntityResourceAssembler assembler) throws URISyntaxException {
+
+        List<Process> processes = uriToEntityConversionService.convertLinks(incoming.getLinks(), Process.class);
+        List<Process> unlinkedProcesses = new ArrayList<>();
+        if (requestMethod.equals(HttpMethod.POST)) {
+            processes.forEach(process -> {
+                biomaterial.addAsInputToProcess(process);
+            });
+        } else if (requestMethod.equals(HttpMethod.PUT)) {
+            unlinkedProcesses = new ArrayList(Arrays.asList(biomaterial.getInputToProcesses().toArray()));
+            biomaterial.getInputToProcesses().clear();
+            biomaterial.getInputToProcesses().addAll(processes);
+        }
+
+        biomaterialRepository.save(biomaterial);
+
+        List<MetadataDocument> metadataToSetToDraft = new ArrayList<>();
+        metadataToSetToDraft.addAll(unlinkedProcesses);
+        metadataToSetToDraft.addAll(processes);
+        metadataToSetToDraft.add(biomaterial);
+        metadataToSetToDraft.forEach(metadataDocument -> {
+            validationStateChangeService.changeValidationState(metadataDocument.getType(), metadataDocument.getId(), ValidationState.DRAFT);
+        });
+
+        return ResponseEntity.ok().build();
+    }
+
+    @RequestMapping(path = "/biomaterials/{id}/derivedByProcesses", method = {PUT, POST}, consumes = {TEXT_URI_LIST_VALUE})
+    HttpEntity<?> overrideLinkBiomaterialAsDerivedByProcessesDefaultEndpoint(@PathVariable("id") Biomaterial biomaterial,
```

Comment:
Most of the code is the same in this method and in the previous one (`overrideLinkBiomaterialAsInputToProcessesDefaultEndpoint`). Could you DRY it, please? Tx